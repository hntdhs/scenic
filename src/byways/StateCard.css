#state {
    font-family: fantasy;
}

.state-name {
    font-family: fantasy;
}