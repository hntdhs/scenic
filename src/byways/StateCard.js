import React from "react";
import { Link } from "react-router-dom";
import "./StateCard.css"


function StateCard ({ name, nickname, image, className }) {
    return (
        <span class='state' className={className}>
            <Link id={name} to={`/states/${name}`} class="link-offset-2 link-underline link-underline-opacity-0">
                <h3 className="state-name">{name}</h3>
                {image}
                <h4 className="state-nickname">{nickname}</h4>
            </Link>
        </span>
    );
}

// function StateCard ({ name, nickname, image }) {
//     return (
//         <span class='state'>
//             <h3><Link id={name} to={`/states/${name}`}>{name}</Link></h3>
//             {image}
//             <h4>{nickname}</h4>
//         </span>
//     );
// }

export default StateCard;