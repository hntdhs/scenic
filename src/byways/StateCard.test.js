import React from "react";
import { render } from "@testing-library/react";
import StateCard from "./StateCard";
import { MemoryRouter } from "react-router";
import { expect } from "@jest/globals";

it("matches snapshot", function () {
    const stateCard = { name: "Nebraska", image: "/stateOutlines/Nebraska.png", nickname: "Cornhusker State"  }
    const { asFragment } = render(
        <MemoryRouter>
            <StateCard stateCard={stateCard} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});