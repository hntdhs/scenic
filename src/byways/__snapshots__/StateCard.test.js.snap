// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/states/undefined"
  >
    <h3 />
    <h4 />
  </a>
</DocumentFragment>
`;
